### ISSUE TITLE
Use 'proper' rotation instead of cholesky decomposition?

### ISSUE THREAD

**Issue by scheidan** [scheidan/BarkerMCMC.jl#4]
As the proposals are non-Gaussian, introducing the correlation via cholesky decomposition is not optimal, see e.g. https://stats.stackexchange.com/questions/83850/confused-about-cholesky-and-eigen-decomposition

A 'real' rotation of the proposal may be more appropriate:

z' =  R * S * z

where R are the eigen vectors of the convariance matrix, and S a diagonal matrix containing the _square root_ of the corresponding eigen values.

A eigen value decomposition, however, is more expensive.

**Comment by scheidan** [comment]
Demo:
```Julia
using LinearAlgebra
using Statistics
using Plots

# covariance matirx
Σ = [1 0.8;
     0.8  1.0]

## not difference for Gaussian
# z = randn(2, 5000) ./ 5
z = (rand(2, 5000) .- 0.5)

C = cholesky(Σ)
x_chol = C.L * z

V, R = eigen(Σ)
S = diagm(sqrt.(V))
x_eigen = R * S * z

# same covariance...
cor(x_chol')
cor(x_eigen')

# ... but different distributions
plot(
    scatter(x_chol[1,:], x_chol[2,:], title="cholesky decomp."),
    scatter(x_eigen[1,:], x_eigen[2,:], title="eigen value decomp."),
    aspect_ratio=1, legend=false,
    xlim=(-1, 1), ylim=(-1, 1)
)
```
![screenshot](https://user-images.githubusercontent.com/3071486/162390419-02a017a2-397f-4a6d-8fcb-1cb4ec398a36.png)

**Comment by scheidan** [comment]
implemented by [5d79198](https://github.com/scheidan/BarkerMCMC.jl/commit/5d79198f6133e318dc41bab002ea750d878253e2)
